pkg/pluginmanager/manager.go

```diff
@@ -737,7 +741,25 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
 	}
 
 	// Log message based on different installation conditions
-	logPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
+	installingMsg, installedMsg := getPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
+
+	var spinnerErr error
+	var sow component.OutputWriterSpinner
+
+	// Initialize the spinner if the spinner is allowed
+	if component.IsTTYEnabled() {
+		// Set spinner options
+		spinnerOptions := component.OutputWriterSpinnerOptions{
+			SpinnerOptions: []component.OutputWriterSpinnerOption{
+				component.WithSpinnerFinalMsg(installedMsg, log.LogTypeINFO),
+			},
+		}
+		// Initialize the spinner
+		sow, spinnerErr = component.NewOutputWriterSpinnerWithSpinnerOptions(os.Stdout, component.TableOutputType, installingMsg, true, spinnerOptions)
+		if spinnerErr != nil {
+			log.Info("Unable to initialize spinner to show plugin installation progress")
+		}
+	}
 
 	if plugin == nil {
 		binary, err := fetchAndVerifyPlugin(p, version)
```

Comment:
If we hit either one of the two `return err` below, the plugin will not be installed, but when we stop the spinner it will say the plugin was installed.  I think we should change the finalMsg of the spinner in case of such an error.

Reply:
Updated. thanks @marckhouzam